### ISSUE TITLE
When convert mxnet model (simple pose) to onnx , the deconvolution’s convert function miss?

### ISSUE THREAD

**Issue by SyGoing** [apache/incubator-mxnet#16176]
Hello，It failed when I try to convert the mxnet model(simple pose in the model zoo).
the wrong message is  that :
   AttributeError: No conversion function registered for op type Deconvolution yet.

**Comment by vandanavk** [comment]
@SyGoing Please share which version of MXNet and ONNx you are using. MXNet does have an exporter for Deconvolution https://github.com/apache/incubator-mxnet/blob/3dacabeaa4dea080f91363701a4a928cc807ed54/python/mxnet/contrib/onnx/mx2onnx/_op_translations.py#L222

**Comment by SyGoing** [comment]
@vandanavk，Hello ,the version of  MXNet and ONNX is 1.4.1 and 1.5.0, So Maybe I need to update mxnet to 1.5.0?

**Comment by SyGoing** [comment]
Hello，when I update mxnet to 1.5.0, there is another problem log for Batchnorm :
D:\Anaconda3\python.exe "D:\PyCharm Community Edition 2019.1.3\helpers\pydev\pydevd.py" --multiproc --qt-support=auto --client 127.0.0.1 --port 8034 --file E:/coco/COCO/workspace_mypy/mxnet2onnx.py
pydev debugger: process 25688 is connecting

Connected to pydev debugger (build 191.7479.30)
Backend Qt5Agg is interactive backend. Turning interactive mode on.
INFO:root:Converting idx: 0, op: null, name: data
INFO:root:Converting idx: 1, op: null, name: resnetv1b_conv0_weight
INFO:root:Converting idx: 2, op: Convolution, name: resnetv1b_conv0_fwd
INFO:root:Converting idx: 3, op: null, name: resnetv1b_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 4, op: null, name: resnetv1b_batchnormcudnnoff0_beta
INFO:root:Converting idx: 5, op: null, name: resnetv1b_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 6, op: null, name: resnetv1b_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 7, op: BatchNorm, name: resnetv1b_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 8, op: Activation, name: resnetv1b_relu0_fwd
INFO:root:Converting idx: 9, op: Pooling, name: resnetv1b_pool0_fwd
INFO:root:Converting idx: 10, op: null, name: resnetv1b_layers1_conv0_weight
INFO:root:Converting idx: 11, op: Convolution, name: resnetv1b_layers1_conv0_fwd
INFO:root:Converting idx: 12, op: null, name: resnetv1b_layers1_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 13, op: null, name: resnetv1b_layers1_batchnormcudnnoff0_beta
INFO:root:Converting idx: 14, op: null, name: resnetv1b_layers1_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 15, op: null, name: resnetv1b_layers1_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 16, op: BatchNorm, name: resnetv1b_layers1_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 17, op: Activation, name: resnetv1b_layers1_relu0_fwd
INFO:root:Converting idx: 18, op: null, name: resnetv1b_layers1_conv1_weight
INFO:root:Converting idx: 19, op: Convolution, name: resnetv1b_layers1_conv1_fwd
INFO:root:Converting idx: 20, op: null, name: resnetv1b_layers1_batchnormcudnnoff1_gamma
INFO:root:Converting idx: 21, op: null, name: resnetv1b_layers1_batchnormcudnnoff1_beta
INFO:root:Converting idx: 22, op: null, name: resnetv1b_layers1_batchnormcudnnoff1_running_mean
INFO:root:Converting idx: 23, op: null, name: resnetv1b_layers1_batchnormcudnnoff1_running_var
INFO:root:Converting idx: 24, op: BatchNorm, name: resnetv1b_layers1_batchnormcudnnoff1_fwd
INFO:root:Converting idx: 25, op: elemwise_add, name: resnetv1b_layers1_basicblockv1b0__plus0
INFO:root:Converting idx: 26, op: Activation, name: resnetv1b_layers1_relu1_fwd
INFO:root:Converting idx: 27, op: null, name: resnetv1b_layers1_conv2_weight
INFO:root:Converting idx: 28, op: Convolution, name: resnetv1b_layers1_conv2_fwd
INFO:root:Converting idx: 29, op: null, name: resnetv1b_layers1_batchnormcudnnoff2_gamma
INFO:root:Converting idx: 30, op: null, name: resnetv1b_layers1_batchnormcudnnoff2_beta
INFO:root:Converting idx: 31, op: null, name: resnetv1b_layers1_batchnormcudnnoff2_running_mean
INFO:root:Converting idx: 32, op: null, name: resnetv1b_layers1_batchnormcudnnoff2_running_var
INFO:root:Converting idx: 33, op: BatchNorm, name: resnetv1b_layers1_batchnormcudnnoff2_fwd
INFO:root:Converting idx: 34, op: Activation, name: resnetv1b_layers1_relu2_fwd
INFO:root:Converting idx: 35, op: null, name: resnetv1b_layers1_conv3_weight
INFO:root:Converting idx: 36, op: Convolution, name: resnetv1b_layers1_conv3_fwd
INFO:root:Converting idx: 37, op: null, name: resnetv1b_layers1_batchnormcudnnoff3_gamma
INFO:root:Converting idx: 38, op: null, name: resnetv1b_layers1_batchnormcudnnoff3_beta
INFO:root:Converting idx: 39, op: null, name: resnetv1b_layers1_batchnormcudnnoff3_running_mean
INFO:root:Converting idx: 40, op: null, name: resnetv1b_layers1_batchnormcudnnoff3_running_var
INFO:root:Converting idx: 41, op: BatchNorm, name: resnetv1b_layers1_batchnormcudnnoff3_fwd
INFO:root:Converting idx: 42, op: elemwise_add, name: resnetv1b_layers1_basicblockv1b1__plus0
INFO:root:Converting idx: 43, op: Activation, name: resnetv1b_layers1_relu3_fwd
INFO:root:Converting idx: 44, op: null, name: resnetv1b_layers2_conv0_weight
INFO:root:Converting idx: 45, op: Convolution, name: resnetv1b_layers2_conv0_fwd
INFO:root:Converting idx: 46, op: null, name: resnetv1b_layers2_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 47, op: null, name: resnetv1b_layers2_batchnormcudnnoff0_beta
INFO:root:Converting idx: 48, op: null, name: resnetv1b_layers2_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 49, op: null, name: resnetv1b_layers2_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 50, op: BatchNorm, name: resnetv1b_layers2_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 51, op: Activation, name: resnetv1b_layers2_relu0_fwd
INFO:root:Converting idx: 52, op: null, name: resnetv1b_layers2_conv1_weight
INFO:root:Converting idx: 53, op: Convolution, name: resnetv1b_layers2_conv1_fwd
INFO:root:Converting idx: 54, op: null, name: resnetv1b_layers2_batchnormcudnnoff1_gamma
INFO:root:Converting idx: 55, op: null, name: resnetv1b_layers2_batchnormcudnnoff1_beta
INFO:root:Converting idx: 56, op: null, name: resnetv1b_layers2_batchnormcudnnoff1_running_mean
INFO:root:Converting idx: 57, op: null, name: resnetv1b_layers2_batchnormcudnnoff1_running_var
INFO:root:Converting idx: 58, op: BatchNorm, name: resnetv1b_layers2_batchnormcudnnoff1_fwd
INFO:root:Converting idx: 59, op: null, name: resnetv1b_down2_conv0_weight
INFO:root:Converting idx: 60, op: Convolution, name: resnetv1b_down2_conv0_fwd
INFO:root:Converting idx: 61, op: null, name: resnetv1b_down2_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 62, op: null, name: resnetv1b_down2_batchnormcudnnoff0_beta
INFO:root:Converting idx: 63, op: null, name: resnetv1b_down2_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 64, op: null, name: resnetv1b_down2_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 65, op: BatchNorm, name: resnetv1b_down2_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 66, op: elemwise_add, name: resnetv1b_layers2_basicblockv1b0__plus0
INFO:root:Converting idx: 67, op: Activation, name: resnetv1b_layers2_relu1_fwd
INFO:root:Converting idx: 68, op: null, name: resnetv1b_layers2_conv2_weight
INFO:root:Converting idx: 69, op: Convolution, name: resnetv1b_layers2_conv2_fwd
INFO:root:Converting idx: 70, op: null, name: resnetv1b_layers2_batchnormcudnnoff2_gamma
INFO:root:Converting idx: 71, op: null, name: resnetv1b_layers2_batchnormcudnnoff2_beta
INFO:root:Converting idx: 72, op: null, name: resnetv1b_layers2_batchnormcudnnoff2_running_mean
INFO:root:Converting idx: 73, op: null, name: resnetv1b_layers2_batchnormcudnnoff2_running_var
INFO:root:Converting idx: 74, op: BatchNorm, name: resnetv1b_layers2_batchnormcudnnoff2_fwd
INFO:root:Converting idx: 75, op: Activation, name: resnetv1b_layers2_relu2_fwd
INFO:root:Converting idx: 76, op: null, name: resnetv1b_layers2_conv3_weight
INFO:root:Converting idx: 77, op: Convolution, name: resnetv1b_layers2_conv3_fwd
INFO:root:Converting idx: 78, op: null, name: resnetv1b_layers2_batchnormcudnnoff3_gamma
INFO:root:Converting idx: 79, op: null, name: resnetv1b_layers2_batchnormcudnnoff3_beta
INFO:root:Converting idx: 80, op: null, name: resnetv1b_layers2_batchnormcudnnoff3_running_mean
INFO:root:Converting idx: 81, op: null, name: resnetv1b_layers2_batchnormcudnnoff3_running_var
INFO:root:Converting idx: 82, op: BatchNorm, name: resnetv1b_layers2_batchnormcudnnoff3_fwd
INFO:root:Converting idx: 83, op: elemwise_add, name: resnetv1b_layers2_basicblockv1b1__plus0
INFO:root:Converting idx: 84, op: Activation, name: resnetv1b_layers2_relu3_fwd
INFO:root:Converting idx: 85, op: null, name: resnetv1b_layers3_conv0_weight
INFO:root:Converting idx: 86, op: Convolution, name: resnetv1b_layers3_conv0_fwd
INFO:root:Converting idx: 87, op: null, name: resnetv1b_layers3_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 88, op: null, name: resnetv1b_layers3_batchnormcudnnoff0_beta
INFO:root:Converting idx: 89, op: null, name: resnetv1b_layers3_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 90, op: null, name: resnetv1b_layers3_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 91, op: BatchNorm, name: resnetv1b_layers3_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 92, op: Activation, name: resnetv1b_layers3_relu0_fwd
INFO:root:Converting idx: 93, op: null, name: resnetv1b_layers3_conv1_weight
INFO:root:Converting idx: 94, op: Convolution, name: resnetv1b_layers3_conv1_fwd
INFO:root:Converting idx: 95, op: null, name: resnetv1b_layers3_batchnormcudnnoff1_gamma
INFO:root:Converting idx: 96, op: null, name: resnetv1b_layers3_batchnormcudnnoff1_beta
INFO:root:Converting idx: 97, op: null, name: resnetv1b_layers3_batchnormcudnnoff1_running_mean
INFO:root:Converting idx: 98, op: null, name: resnetv1b_layers3_batchnormcudnnoff1_running_var
INFO:root:Converting idx: 99, op: BatchNorm, name: resnetv1b_layers3_batchnormcudnnoff1_fwd
INFO:root:Converting idx: 100, op: null, name: resnetv1b_down3_conv0_weight
INFO:root:Converting idx: 101, op: Convolution, name: resnetv1b_down3_conv0_fwd
INFO:root:Converting idx: 102, op: null, name: resnetv1b_down3_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 103, op: null, name: resnetv1b_down3_batchnormcudnnoff0_beta
INFO:root:Converting idx: 104, op: null, name: resnetv1b_down3_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 105, op: null, name: resnetv1b_down3_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 106, op: BatchNorm, name: resnetv1b_down3_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 107, op: elemwise_add, name: resnetv1b_layers3_basicblockv1b0__plus0
INFO:root:Converting idx: 108, op: Activation, name: resnetv1b_layers3_relu1_fwd
INFO:root:Converting idx: 109, op: null, name: resnetv1b_layers3_conv2_weight
INFO:root:Converting idx: 110, op: Convolution, name: resnetv1b_layers3_conv2_fwd
INFO:root:Converting idx: 111, op: null, name: resnetv1b_layers3_batchnormcudnnoff2_gamma
INFO:root:Converting idx: 112, op: null, name: resnetv1b_layers3_batchnormcudnnoff2_beta
INFO:root:Converting idx: 113, op: null, name: resnetv1b_layers3_batchnormcudnnoff2_running_mean
INFO:root:Converting idx: 114, op: null, name: resnetv1b_layers3_batchnormcudnnoff2_running_var
INFO:root:Converting idx: 115, op: BatchNorm, name: resnetv1b_layers3_batchnormcudnnoff2_fwd
INFO:root:Converting idx: 116, op: Activation, name: resnetv1b_layers3_relu2_fwd
INFO:root:Converting idx: 117, op: null, name: resnetv1b_layers3_conv3_weight
INFO:root:Converting idx: 118, op: Convolution, name: resnetv1b_layers3_conv3_fwd
INFO:root:Converting idx: 119, op: null, name: resnetv1b_layers3_batchnormcudnnoff3_gamma
INFO:root:Converting idx: 120, op: null, name: resnetv1b_layers3_batchnormcudnnoff3_beta
INFO:root:Converting idx: 121, op: null, name: resnetv1b_layers3_batchnormcudnnoff3_running_mean
INFO:root:Converting idx: 122, op: null, name: resnetv1b_layers3_batchnormcudnnoff3_running_var
INFO:root:Converting idx: 123, op: BatchNorm, name: resnetv1b_layers3_batchnormcudnnoff3_fwd
INFO:root:Converting idx: 124, op: elemwise_add, name: resnetv1b_layers3_basicblockv1b1__plus0
INFO:root:Converting idx: 125, op: Activation, name: resnetv1b_layers3_relu3_fwd
INFO:root:Converting idx: 126, op: null, name: resnetv1b_layers4_conv0_weight
INFO:root:Converting idx: 127, op: Convolution, name: resnetv1b_layers4_conv0_fwd
INFO:root:Converting idx: 128, op: null, name: resnetv1b_layers4_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 129, op: null, name: resnetv1b_layers4_batchnormcudnnoff0_beta
INFO:root:Converting idx: 130, op: null, name: resnetv1b_layers4_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 131, op: null, name: resnetv1b_layers4_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 132, op: BatchNorm, name: resnetv1b_layers4_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 133, op: Activation, name: resnetv1b_layers4_relu0_fwd
INFO:root:Converting idx: 134, op: null, name: resnetv1b_layers4_conv1_weight
INFO:root:Converting idx: 135, op: Convolution, name: resnetv1b_layers4_conv1_fwd
INFO:root:Converting idx: 136, op: null, name: resnetv1b_layers4_batchnormcudnnoff1_gamma
INFO:root:Converting idx: 137, op: null, name: resnetv1b_layers4_batchnormcudnnoff1_beta
INFO:root:Converting idx: 138, op: null, name: resnetv1b_layers4_batchnormcudnnoff1_running_mean
INFO:root:Converting idx: 139, op: null, name: resnetv1b_layers4_batchnormcudnnoff1_running_var
INFO:root:Converting idx: 140, op: BatchNorm, name: resnetv1b_layers4_batchnormcudnnoff1_fwd
INFO:root:Converting idx: 141, op: null, name: resnetv1b_down4_conv0_weight
INFO:root:Converting idx: 142, op: Convolution, name: resnetv1b_down4_conv0_fwd
INFO:root:Converting idx: 143, op: null, name: resnetv1b_down4_batchnormcudnnoff0_gamma
INFO:root:Converting idx: 144, op: null, name: resnetv1b_down4_batchnormcudnnoff0_beta
INFO:root:Converting idx: 145, op: null, name: resnetv1b_down4_batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 146, op: null, name: resnetv1b_down4_batchnormcudnnoff0_running_var
INFO:root:Converting idx: 147, op: BatchNorm, name: resnetv1b_down4_batchnormcudnnoff0_fwd
INFO:root:Converting idx: 148, op: elemwise_add, name: resnetv1b_layers4_basicblockv1b0__plus0
INFO:root:Converting idx: 149, op: Activation, name: resnetv1b_layers4_relu1_fwd
INFO:root:Converting idx: 150, op: null, name: resnetv1b_layers4_conv2_weight
INFO:root:Converting idx: 151, op: Convolution, name: resnetv1b_layers4_conv2_fwd
INFO:root:Converting idx: 152, op: null, name: resnetv1b_layers4_batchnormcudnnoff2_gamma
INFO:root:Converting idx: 153, op: null, name: resnetv1b_layers4_batchnormcudnnoff2_beta
INFO:root:Converting idx: 154, op: null, name: resnetv1b_layers4_batchnormcudnnoff2_running_mean
INFO:root:Converting idx: 155, op: null, name: resnetv1b_layers4_batchnormcudnnoff2_running_var
INFO:root:Converting idx: 156, op: BatchNorm, name: resnetv1b_layers4_batchnormcudnnoff2_fwd
INFO:root:Converting idx: 157, op: Activation, name: resnetv1b_layers4_relu2_fwd
INFO:root:Converting idx: 158, op: null, name: resnetv1b_layers4_conv3_weight
INFO:root:Converting idx: 159, op: Convolution, name: resnetv1b_layers4_conv3_fwd
INFO:root:Converting idx: 160, op: null, name: resnetv1b_layers4_batchnormcudnnoff3_gamma
INFO:root:Converting idx: 161, op: null, name: resnetv1b_layers4_batchnormcudnnoff3_beta
INFO:root:Converting idx: 162, op: null, name: resnetv1b_layers4_batchnormcudnnoff3_running_mean
INFO:root:Converting idx: 163, op: null, name: resnetv1b_layers4_batchnormcudnnoff3_running_var
INFO:root:Converting idx: 164, op: BatchNorm, name: resnetv1b_layers4_batchnormcudnnoff3_fwd
INFO:root:Converting idx: 165, op: elemwise_add, name: resnetv1b_layers4_basicblockv1b1__plus0
INFO:root:Converting idx: 166, op: Activation, name: resnetv1b_layers4_relu3_fwd
INFO:root:Converting idx: 167, op: null, name: conv0_weight
INFO:root:Converting idx: 168, op: Deconvolution, name: conv0_fwd
INFO:root:Converting idx: 169, op: null, name: batchnormcudnnoff0_gamma
INFO:root:Converting idx: 170, op: null, name: batchnormcudnnoff0_beta
INFO:root:Converting idx: 171, op: null, name: batchnormcudnnoff0_running_mean
INFO:root:Converting idx: 172, op: null, name: batchnormcudnnoff0_running_var
INFO:root:Converting idx: 173, op: BatchNorm, name: batchnormcudnnoff0_fwd
INFO:root:Converting idx: 174, op: Activation, name: relu0_fwd
INFO:root:Converting idx: 175, op: null, name: conv1_weight
INFO:root:Converting idx: 176, op: Deconvolution, name: conv1_fwd
INFO:root:Converting idx: 177, op: null, name: batchnormcudnnoff1_gamma
INFO:root:Converting idx: 178, op: null, name: batchnormcudnnoff1_beta
INFO:root:Converting idx: 179, op: null, name: batchnormcudnnoff1_running_mean
INFO:root:Converting idx: 180, op: null, name: batchnormcudnnoff1_running_var
INFO:root:Converting idx: 181, op: BatchNorm, name: batchnormcudnnoff1_fwd
INFO:root:Converting idx: 182, op: Activation, name: relu1_fwd
INFO:root:Converting idx: 183, op: null, name: conv2_weight
INFO:root:Converting idx: 184, op: Deconvolution, name: conv2_fwd
INFO:root:Converting idx: 185, op: null, name: batchnormcudnnoff2_gamma
INFO:root:Converting idx: 186, op: null, name: batchnormcudnnoff2_beta
INFO:root:Converting idx: 187, op: null, name: batchnormcudnnoff2_running_mean
INFO:root:Converting idx: 188, op: null, name: batchnormcudnnoff2_running_var
INFO:root:Converting idx: 189, op: BatchNorm, name: batchnormcudnnoff2_fwd
INFO:root:Converting idx: 190, op: Activation, name: relu2_fwd
INFO:root:Converting idx: 191, op: null, name: conv3_weight
INFO:root:Converting idx: 192, op: null, name: conv3_bias
INFO:root:Converting idx: 193, op: Convolution, name: conv3_fwd
INFO:root:Output node is: conv3_fwd
Traceback (most recent call last):
  File "D:\PyCharm Community Edition 2019.1.3\helpers\pydev\pydevd.py", line 1758, in <module>
    main()
  File "D:\PyCharm Community Edition 2019.1.3\helpers\pydev\pydevd.py", line 1752, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "D:\PyCharm Community Edition 2019.1.3\helpers\pydev\pydevd.py", line 1147, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "D:\PyCharm Community Edition 2019.1.3\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "E:/coco/COCO/workspace_mypy/mxnet2onnx.py", line 58, in <module>
    generate_onnx_file('test-symbol.json','test-0000.params','simple_base.onnx')
  File "E:/coco/COCO/workspace_mypy/mxnet2onnx.py", line 47, in generate_onnx_file
    onnx_mxnet.export_model(net_symbol, net_params, [input_shape], numpy.float32, onnx_path, verbose=True)
  File "D:\Anaconda3\lib\site-packages\mxnet\contrib\onnx\mx2onnx\export_model.py", line 87, in export_model
    verbose=verbose)
  File "D:\Anaconda3\lib\site-packages\mxnet\contrib\onnx\mx2onnx\export_onnx.py", line 312, in create_onnx_graph_proto
    checker.check_graph(graph)
  File "D:\Anaconda3\lib\site-packages\onnx\checker.py", line 52, in checker
    proto.SerializeToString(), ctx)
onnx.onnx_cpp2py_export.checker.ValidationError: Unrecognized attribute: spatial for operator BatchNormalization

==> Context: Bad node spec: input: "resnetv1b_conv0_fwd" input: "resnetv1b_batchnormcudnnoff0_gamma" input: "resnetv1b_batchnormcudnnoff0_beta" input: "resnetv1b_batchnormcudnnoff0_running_mean" input: "resnetv1b_batchnormcudnnoff0_running_var" output: "resnetv1b_batchnormcudnnoff0_fwd" name: "resnetv1b_batchnormcudnnoff0_fwd" op_type: "BatchNormalization" attribute { name: "epsilon" f: 1e-05 type: FLOAT } attribute { name: "momentum" f: 0.9 type: FLOAT } attribute { name: "spatial" i: 0 type: INT }

Process finished with exit code 1

**Comment by SyGoing** [comment]
my code  is here:

import mxnet
import mxnet  as mx
from mxnet.contrib import onnx as onnx_mxnet
from onnx import checker
import onnx


# convert pretrained glunCV2jsonparam
from gluoncv import model_zoo
from gluoncv.utils import export_block
from mxnet.gluon.model_zoo import vision

def gen_symbol_param_file_form_gluoncv_test():
    pose_net = model_zoo.get_model('simple_pose_resnet18_v1b', pretrained=True)
    x = mxnet.ndarray.random.normal(shape=(1, 3,192,256))
    pose_net.hybridize()
    predicted_heatmap=pose_net(x)
    pose_net.export("test")


def generate_onnx_file(symbol_path,param_path,onnx_path,input_shape=(1,3,192,256)):
    logging.basicConfig(level=logging.INFO)

    net_symbol = mxnet.symbol.load(symbol_path)
    net_params_raw = mxnet.nd.load(param_path)
    net_params = dict()
    for k, v in net_params_raw.items():
        tp, name = k.split(':', 1)
        net_params.update({name: v})

    onnx_mxnet.export_model(net_symbol, net_params, [input_shape], numpy.float32, onnx_path, verbose=True)

    # Load onnx model
    model_proto = onnx.load_model(onnx_path)

    # Check if converted ONNX protobuf is valid
    checker.check_graph(model_proto.graph)


if __name__ == '__main__':
    gen_symbol_param_file_form_gluoncv_test()
    generate_onnx_file('test-symbol.json','test-0000.params','simple_base.onnx')

**Comment by Rainweic** [comment]
> Unrecognized attribute: spatial for operator BatchNormalization

``Unrecognized attribute: spatial for operator BatchNormalization`` 
you use onnx 1.3.0 can solve this

I use mxnet1.5.1 onnx1.3.0 , and meet ``AttributeError: No conversion function registered for op type _greater_scalar yet.``

**Comment by photoszzt** [comment]
I'm hitting this problem:
I'm using Mxnet 1.6.0 and onnx 1.7.0. 
```
onnx.onnx_cpp2py_export.checker.ValidationError: Unrecognized attribute: spatial for operator BatchNormalization

==> Context: Bad node spec: input: "resnetv1b_conv0_fwd" input: "resnetv1b_batchnormcudnnoff0_gamma" input: "resnetv1b_batchnormcudnnoff0_beta" input: "resnetv1b_batchnormcudnnoff0_running_mean" input: "resnetv1b_batchnormcudnnoff0_running_var" output: "resnetv1b_batchnormcudnnoff0_fwd" name: "resnetv1b_batchnormcudnnoff0_fwd" op_type: "BatchNormalization" attribute { name: "epsilon" f: 1e-05 type: FLOAT } attribute { name: "momentum" f: 0.9 type: FLOAT } attribute { name: "spatial" i: 0 type: INT }
```

**Comment by TristonC** [comment]
@Zha0q1 @szha Have this issue been solved in 1.x?

**Comment by Zha0q1** [comment]
decovolution should be supported now! 

You can refer to this latest op support matrix https://github.com/apache/incubator-mxnet/tree/v1.x/python/mxnet/onnx#operator-support-matrix. The new mx2onnx export module will be released soon with mxnet 1.9!

Also https://github.com/apache/incubator-mxnet/tree/v1.x/python/mxnet/onnx#gluoncv-pretrained-model-support-matrix here's the pretrained model support for cv and nlp

**Comment by MrYANG23** [comment]
> I'm hitting this problem:
> I'm using Mxnet 1.6.0 and onnx 1.7.0.
> 
> ```
> onnx.onnx_cpp2py_export.checker.ValidationError: Unrecognized attribute: spatial for operator BatchNormalization
> 
> ==> Context: Bad node spec: input: "resnetv1b_conv0_fwd" input: "resnetv1b_batchnormcudnnoff0_gamma" input: "resnetv1b_batchnormcudnnoff0_beta" input: "resnetv1b_batchnormcudnnoff0_running_mean" input: "resnetv1b_batchnormcudnnoff0_running_var" output: "resnetv1b_batchnormcudnnoff0_fwd" name: "resnetv1b_batchnormcudnnoff0_fwd" op_type: "BatchNormalization" attribute { name: "epsilon" f: 1e-05 type: FLOAT } attribute { name: "momentum" f: 0.9 type: FLOAT } attribute { name: "spatial" i: 0 type: INT }
> ```

did you solved the problem ,i have the same issues
